File: 2021/10/19/main.go
package main

func nextGreaterElement(nums1 []int, nums2 []int) []int {
	l := make(map[int]int)
	for i, n := range nums2 {
		l[n] = i
	}

	r := make([]int, len(nums1))
	for i, n := range nums1 {
		r[i] = -1
		for j := l[n] + 1; j < len(nums2); j++ {
			if n < nums2[j] {
				r[i] = nums2[j]
				break
			}
		}
	}

	return r
}

File: 2021/10/19/main.py
class Solution:
    def nextGreaterElement(self, nums1: List[int], nums2: List[int]) -> List[int]:
        r, l = [], {}
        for i, n in enumerate(nums2):
            l[n] = i
        for i, n in enumerate(nums1):
            r.append(-1)
            for j in range(l[n]+1, len(nums2)):
                if n < nums2[j]:
                    r[i] = nums2[j]
                    break
        return r